src/lib/encoder.ts

```diff
@@ -367,7 +387,7 @@ const decodeBytesNCompactBytesArray = (
 ): string[] => {
   const bytesValues = decodeCompactBytesArray(compactBytesArray, consumed);
   bytesValues.forEach((bytesValue, index) => {
-    if (stripHexPrefix(bytesValue).length > numberOfBytes * 2)
+    if (bytesValue.slice(2).length > numberOfBytes * 2)
```

Comment:
I think since there is no `stripHexPrefix` function in viem, we could create our own local utility function in here? That would make more sense codewise when reading these `<...>.slice(2)` functions